### ISSUE TITLE
linter: allow specification of jsPlugins by reference

### ISSUE THREAD

**Issue by cellison-figma** [oxc-project/oxc#23944]
### What version of Oxlint are you using?

1.67.0

### What command did you run?

oxlint -c oxlint.config.ts

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

In PNPM workspace with dir `packages` containing shared workspace packages (simplified for report)

```ts
// packages/oxlint/oxlint.config.base.ts
import { defineConfig } from 'oxlint'

export default defineConfig({
  jsPlugins: ['eslint-plugin-simple-import-sort'],
})
```

```ts
// packages/core/oxlint.config.ts <- package.json required to specify `eslint-plugin-simple-import-sort` as a direct dep
import { defineConfig } from 'oxlint'
import baseConfig from '@internal/oxlint/oxlint.config.base.ts'

export default defineConfig({
  extends: [baseConfig],
})
```

### What happened?

Oxlint JsPlugins can only be specified by import specifier. This deviates from the eslint plugin configuration spec, which allows for plugin specification by reference. See example below.

```js
// eslint.config.js
import example from "eslint-plugin-example";
import { defineConfig } from "eslint/config";

export default defineConfig([
	{
		plugins: {
			example,
		},
		rules: {
			"example/rule1": "warn",
		},
	},
]);
```
> from [configure plugins section](https://eslint.org/docs/latest/use/configure/plugins#configure-plugins) of eslint docs

This presents an issue, as base configurations cannot properly encapsulate their dependencies. A config that extends another with JS Plugins is required to list all included plugins as 1P deps, rather than inheriting them transitively. This is unnecessary package.json bloat and violates intentional package boundaries, as consumers could inadvertently install different plugin versions or otherwise become out of sync with the intended rule provider.

The equivalent should be possible in oxlint, wherein a base config could specify the following:

```ts
// packages/oxlint/oxlint.config.base.ts
import simpleImportSortPlugin from 'eslint-plugin-simple-import-sort'
import { defineConfig } from 'oxlint'

export default defineConfig({
  jsPlugins: [
    { 
      name: 'simple-import-sort',
      // perhaps overload the type of `specifier` or introduce some other key name option here
      plugin: simpleImportSortPlugin
    }
  ]
  // allows ref of rules via `simple-import-sort` name
})
```

Consumers extending off that config should then not have to list `eslint-plugin-simple-import-sort` as a direct dependency, as the exporting package handles resolution.

Happy to reclassify if Bug is not correct, but the API for specifying plugins does not match that of eslint.

**Comment by camc314** [comment]
cc @overlookmotel mind having a look at this? Iirc, it's due to the potential for multithreading right?

**Comment by marcoreni** [comment]
Any updates about this? We're also impacted

**Comment by patricktree** [comment]
A workaround is to use `import.meta.resolve` inside the shared config, so plugins resolve from its own dependencies:

```js
jsPlugins: [
  {
    name: "turbo",
    specifier: import.meta.resolve("eslint-plugin-turbo"),
  },
],
```

Verified with Oxlint 1.72.0 in a pnpm workspace with hoisting disabled: [reproduction → workaround diff](https://github.com/patricktree/oxc-issue-23944-faulty-jsplugins-nodemodules-resolution/compare/main...workaround/resolve-jsplugins-from-shared-config).

Credit to [nkzw-tech/oxlint-config#2](https://github.com/nkzw-tech/oxlint-config/pull/2) for the workaround.

**Comment by cellison-figma** [comment]
The workaround above does sufficiently resolve this issue. Closing.
